prefect_kubernetes/jobs.py

```diff
@@ -0,0 +1,299 @@
+
+    Args:
+        job_name: The name of a job to delete.
+        kubernetes_credentials: `KubernetesCredentials` block
+            holding authentication needed to generate the required API client.
+        body: A Kubernetes `V1DeleteOptions` object.
+        namespace: The Kubernetes namespace to delete this job in.
+        **kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+
+    Returns:
+        A Kubernetes `V1Status` object.
+
+    Example:
+        Delete "my-job" in the default namespace:
+        ```python
+        from kubernetes.client.models import V1DeleteOptions
+        from prefect import flow
+        from prefect_kubernetes.credentials import KubernetesCredentials
+        from prefect_kubernetes.jobs import delete_namespaced_job
+
+        @flow
+        def kubernetes_orchestrator():
+            v1_job_status = delete_namespaced_job(
+                job_name="my-job",
+                kubernetes_credentials=KubernetesCredentials.load("k8s-creds"),
+                body=V1DeleteOptions(propagation_policy="Foreground"),
+            )
+        ```
+    """
+
+    api_client = kubernetes_credentials.get_batch_client()
+
+    return await run_sync_in_worker_thread(
+        api_client.delete_namespaced_job,
+        name=job_name,
+        body=body,
+        namespace=namespace,
+        **kube_kwargs,
+    )
+
+
+@task
+async def list_namespaced_job(
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    **kube_kwargs: Optional[Dict[str, Any]],
+) -> V1JobList:
+    """Task for listing namespaced Kubernetes jobs.
+
+    Args:
+        kubernetes_credentials: `KubernetesCredentials` block
+            holding authentication needed to generate the required API client.
+        namespace: The Kubernetes namespace to list jobs from.
+        **kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+    Returns:
+        A Kubernetes `V1JobList` object.
+
+    Example:
+        List jobs in "my-namespace":
+        ```python
+        from prefect import flow
+        from prefect_kubernetes.credentials import KubernetesCredentials
+        from prefect_kubernetes.jobs import list_namespaced_job
+
+        @flow
+        def kubernetes_orchestrator():
+            namespaced_job_list = list_namespaced_job(
+                namespace="my-namespace",
+                kubernetes_credentials=KubernetesCredentials.load("k8s-creds"),
+            )
+        ```
+    """
+    api_client = kubernetes_credentials.get_batch_client()
+
+    return await run_sync_in_worker_thread(
+        api_client.list_namespaced_job,
+        namespace=namespace,
+        **kube_kwargs,
+    )
+
+
+@task
+async def patch_namespaced_job(
+    job_name: str,
+    body: V1Job,
+    kubernetes_credentials: KubernetesCredentials = None,
```

Comment:
I think KubernetesCredentials. Also, I noticed that the args ordering across tasks is inconsistent; can we order it by:
1. `kubernetes_credentials`
2. `job_name`
3. `body`